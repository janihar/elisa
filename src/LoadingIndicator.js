import React from "react";
import { View, ActivityIndicator, Text } from "react-native";
import PropTypes from "prop-types";

const LoadingIndicator = ({ text }) => {
  return (
    <View>
      <ActivityIndicator size="large" />
      <Text style={{ alignItems: "center", textAlign: "center" }}>{text}</Text>
    </View>
  );
};

LoadingIndicator.propTypes = {
  name: PropTypes.string
};

export default LoadingIndicator;
