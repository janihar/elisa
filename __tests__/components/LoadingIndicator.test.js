import React from "react";
import renderer from "react-test-renderer";
import LoadingIndicator from "../../src/LoadingIndicator";

describe("<LoadingIndicator text={test}/>", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<LoadingIndicator test={"test"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
