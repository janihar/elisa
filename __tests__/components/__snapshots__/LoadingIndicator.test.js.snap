// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator text={test}/> renders correctly 1`] = `
<View>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    style={
      Object {
        "alignItems": "center",
        "textAlign": "center",
      }
    }
  />
</View>
`;
